### PR TITLE
Tech CORE-306 | Create the alias commands on the CLI

### PR DESCRIPTION
**Pull request by luiz-couto**
# Description

Kind: Feature

Section: CLI and client controllers

Summary: 
* Added alias to all commands
* Added alias to client controller

Issue link: https://inspr.atlassian.net/jira/software/projects/CORE/boards/44?selectedIssue=CORE-306

## Changelog
* Apply, describe, get and delete alias
* mocks and tests
* alias client controller

## Pay attention to

> The alias commands

### CODE REVIEW

File: cmd/insprd/operators/kafka/channels/client.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func init() {
 
 // ChannelOperator is a client for channel operations on kafka
 type ChannelOperator struct {
-	k      *kafka.AdminClient
+	k      kafkaAdminClient
```
Comment (brenordr) [inline cmd/insprd/operators/kafka/channels/client.go:25]: @luiz-couto just curious. Why this change?

**Review by brenordr — COMMENTED** [review]
I'll double check tomorrow morning. Seems good from a quick check

File: cmd/insprd/operators/kafka/channels/client.go [hunk 2]
```diff
@@ -155,3 +154,23 @@ func (c *ChannelOperator) Delete(ctx context.Context, context string, name strin
...
 }
+
+type kafkaAdminClient interface {
+	DeleteTopics(ctx context.Context, topics []string, options ...kafka.DeleteTopicsAdminOption) (result []kafka.TopicResult, err error)
+	CreateTopics(ctx context.Context, topics []kafka.TopicSpecification, options ...kafka.CreateTopicsAdminOption) (result []kafka.TopicResult, err error)
+	GetMetadata(topic *string, allTopics bool, timeoutMs int) (*kafka.Metadata, error)
+}
+
+type mockAdminClient struct {
+}
```
Comment (nicholaspcr) [inline cmd/insprd/operators/kafka/channels/client.go:165]: Don't know if it's a good idea or not but it may be interesting to do a `err` field, in which the person using the mock can determine if they want an error or  not in the response.

File: cmd/inspr/cli/apply_alias.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		flagIsUpdate := cmd.InsprOptions.Update
+
+		var log diff.Changelog
+
+		scope, err := cliutils.GetScope()
+		if err != nil {
+			return err
+		}
+
+		parentPath, err := metautils.JoinScopes(scope, alias.Meta.Parent)
```
Comment (franciscobonand) [inline cmd/inspr/cli/apply_alias.go:35]: Won't the 'scope' already be a path to the alias' parent dApp?

File: pkg/meta/utils/parser/alias_yaml.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/inspr/inspr/pkg/meta"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// YamlToAlias - deserializes the yaml to a meta.Alias struct
+func YamlToAlias(bytes []byte) (*meta.Alias, error) {
+	var alias *meta.Alias
+
+	if err := yaml.Unmarshal(bytes, &alias); err != nil {
+		return nil, ierrors.NewError().Message("Error parsing the file").Build()
```
Comment (franciscobonand) [inline pkg/meta/utils/parser/alias_yaml.go]: You can use err.Error( ) as the ierrors message

**Review by franciscobonand — CHANGES_REQUESTED** [review]

File: pkg/meta/utils/parser/alias_yaml.go (hunk 4 above)
Comment (luiz-couto) [reply pkg/meta/utils/parser/alias_yaml.go]: Good idea

File: cmd/inspr/cli/apply_alias.go (hunk 3 above)
Comment (luiz-couto) [reply cmd/inspr/cli/apply_alias.go:35]: I think not. For example, if you have a app1 inside the tree root (and you want to apply the alias inside app1), you could have a scope = "" and the parent of the alias = "app1".

File: cmd/insprd/operators/kafka/channels/client.go (hunk 2 above)
Comment (nicholaspcr) [reply cmd/insprd/operators/kafka/channels/client.go:165]: not really in the scope of your ticket/issue, probably will be done when the need for it comes

**Review by nicholaspcr — APPROVED** [review]

File: cmd/insprd/operators/kafka/channels/client.go (hunk 1 above)
Comment (luiz-couto) [reply cmd/insprd/operators/kafka/channels/client.go:25]: I don't know haha I did not changed that. This diff was a bug when moving the PR from gitlab to github. I'll check if this change is in develop or if I should use the pointer instead

Comment (luiz-couto) [reply cmd/insprd/operators/kafka/channels/client.go:25]: I've looked and it actually was a change that @ptcar2009 did to allow the kafka admin client to be mocked. Now, the kafkaAdminClient is an interface.

**Review by franciscobonand — APPROVED** [review]
